### DISCUSSION TITLE
App Insights - Zaciemnianie/Obfuskacja? danych wrażliwych w logach

### DISCUSSION THREAD

**Discussion by meanin** [AzureCommunityPL/group#147] [Pomoc] [answered]
Aplikacje webowe pisane w C#/.NET często korzystają z AppInsights. Czy jest jakiś standardowy sposób na usunięcie danych wrażliwych? Jakiś nuget, czy mechanizm samego Azure'a?

Z mojego krótkiego researchu to wygląda obiecująco:
https://stackoverflow.com/a/53776460

**Comment by mmisztal1980** [comment, marked as answer]
App Insights posiadają tzw. purge api, przeznaczone do takich celów:
 https://docs.microsoft.com/en-us/rest/api/application-insights/components/purge

Niestety wspiera ono zubożoną ilość operatorów względem kusto oraz ma limity (np 50 requestów na godzinę, a na wyniki trzeba czekać do 72h), więc trzeba się nagimnastykować, np:

Wyciągnąć itemIds wpisów, które chesz usunąć z appInsights:
```
traces
| where timestamp > ago(90d)
| where * contains "something" and * contains "something else"
| project itemId
```

A potem skonstruować body do requestu mniej więcej tak:
`POST https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/components/{resourceName}/purge?api-version=2015-05-01`
```json
{
  "table": "traces",
  "filters": [
    {
      "column": "timestamp",
      "operator": ">",
      "value": "2017-09-01T00:00:00"
    },
   {
      "column": "itemId",
      "operator": "in",
      "value": [
       <GUIDY Z QUERY POWYZEJ>
      ]
    }
  ]
}
```

Odpowiedź z API, będzie zawierała Id operacji `purgeId` np: `purge-_2b8a3fdd-90f0-4b5f-9c3a-99c227de7d23`
Można go potem użyć do sprawdzenia statusu operacji, np tak:

```
GET https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/components/{resourceName}/operations/{purgeId}?api-version=2015-05-01
```

Zabawę w uwierzytelnianie pozostawiam Tobie, nie chcę Ci odbierać całej zabawy :P

Oczywiście, najlepiej jest tych danych po prostu nie logować ;)
